backend/metrics-api.py

```diff
@@ -43,6 +47,96 @@ def make_error_response(params, error, status):
     }
     return Response(json.dumps(data, indent=2), status=status, mimetype='application/json')
 
+@app.route('/api/arrival_download', methods=['GET'])
+def download_arrival_data():
+    '''
+    first step in letting users download arrival data
+    based on the filters they have selected in the frontend
+    '''
+
+    args = request.args
+
+    print(f"args.get('variables'): {args.get('variables')}")
+    variables_dict = json.loads(args.get('variables'))
+
+    agency_id = variables_dict.get('agencyId')
+    agency_config = config.get_agency(agency_id)
+    route_id =variables_dict.get('routeId')
+    direction_id = variables_dict.get('directionId', None)
+    starttime = variables_dict.get('startTime', None)
+    endtime = variables_dict.get('endTime', None)
+
+    arrival_df = pd.DataFrame()
+
+    for date_str in variables_dict.get('dates'):
+
+        date_of_interest = datetime.datetime.strptime(date_str,'%Y-%m-%d').date()
+
+        if (starttime is not None and endtime is not None):
+            starttime_param = util.get_timestamp_or_none(date_of_interest, starttime, agency_config.tz)
+            endtime_param = util.get_timestamp_or_none(date_of_interest, endtime, agency_config.tz)
+        else:
+            starttime_param = None
+            endtime_param = None
+
+        history = arrival_history.get_by_date(agency_id=agency_id
```

Comment:
Handle exception if arrival history doesn't exist

Reply:
I added a very basis try and except statement around the arrival history. I was thinking about adding a future "issue" for making it more robust.